docs/core/v3.0.rst

```diff
@@ -137,6 +138,29 @@ provide a framework for community-defined extensions, which can be
 developed and published independently without requiring centralised
 coordination of all specifications.
 
+See `protocol extensions`_ below.
+
+Stability Policy
+----------------
+
+This core protocol specification adheres to adapted Semantic Versioning regarding
+read and write compatibility using the ``MAJOR.MINOR[.PATCH]?`` version
+number format, where the patch number is optional and implicitly ``0``.
+A zarr implementation provides the read and write API by implementing
+this specification, the datasets are considered consumers in this case.
+
+Notably, this excludes extensions, codecs and stores from the compatibility
```

Comment:
Isn't the rest of the document thought trying to elevate stores to something more like a proper extension?

Reply:
The spec defines some operations that stores should support, but ultimately it is up to each implementation to decide how stores are implemented and what interfaces they may use.

Individual implementations could certainly document what stores they support and any relevant stability guarantees.

Currently there is nothing in the spec itself as far as specifying which store to use for a zarr root, e.g. a path to a local directory, an HTTP URL, in-memory storage, path to zip file, etc.  If we standardized a URL syntax or JSON representation for specifying where a zarr root/array is stored, then it would make sense to talk about stability guarantees of that URL/JSON representation.

I suppose this spec could describe mappings from the "store" abstraction to concrete storage mechanisms, like local files, zip file, GCS/S3, etc.  However those mappings would be quite trivial so I'm not sure it is very useful to standardize that without also standardizing some URL syntax/JSON representation for specifying the store.

Reply:
I assume stores are described in the core specification via the storage interface, but the actual supported storage backends might still have ambiguities how the key-layout is mapped to a specific backend. This would need to be described separately, and is not part of the core spec IMO. Therefore I'd rather list stores explicitly here as well, an implementation following v3.0 doesn't necessarily say anything about the store compatibility (e.g. if it can read from zip-stores).

Reply:
> ultimately it is up to each implementation to decide how stores are implemented and what interfaces they may use.

Yeah, I'm not sure about the interfaces of the stores either though the "feature sets" that are listed do tend to be pretty relevant to the rest of the spec. But I think there are also storage issues which are pretty important at this level. For example, `zip my.zarr my.zip` doesn't just work. Zip storage needs to be specified. I think that will be true to some degree for other stores.

> Individual implementations could certainly document what stores they support and any relevant stability guarantees.

Agreed. And I think that's another key role here is to say whether given storage types are MUST, MAY or SHOULD. 

> If we standardized a URL syntax or JSON representation for specifying where a zarr root/array is stored

:+1: which also reminds me that I think of consolidated metadata as a metadata-storage-type.